plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/DeletionVectors.java

```diff
@@ -0,0 +1,270 @@
+    private static final String INLINE_MARKER = "i"; // inline
+
+    private static final CharMatcher ALPHANUMERIC = CharMatcher.inRange('A', 'Z').or(CharMatcher.inRange('a', 'z')).or(CharMatcher.inRange('0', '9')).precomputed();
+
+    private DeletionVectors() {}
+
+    public static RoaringBitmap[] readDeletionVectors(
+            TrinoFileSystem fileSystem,
+            Location location,
+            String storageType,
+            String pathOrInlineDv,
+            OptionalInt offset,
+            int sizeInBytes,
+            long cardinality)
+            throws IOException
+    {
+        if (storageType.equals(UUID_MARKER)) {
+            TrinoInputFile inputFile = fileSystem.newInputFile(location.appendPath(toFileName(pathOrInlineDv)));
+            ByteBuffer buffer = readDeletionVector(inputFile, offset.orElseThrow(), sizeInBytes);
+            RoaringBitmap[] bitmaps = deserializeDeletionVectors(buffer);
+            if (bitmaps.length != cardinality) {
+                // Don't throw an exception because Databricks may report the wrong cardinality when there are many deleted rows
+                log.debug("The number of deleted rows expects %s but got %s", cardinality, bitmaps.length);
+            }
+            return bitmaps;
+        }
+        else if (storageType.equals(INLINE_MARKER) || storageType.equals(PATH_MARKER)) {
+            throw new TrinoException(NOT_SUPPORTED, "Unsupported storage type for deletion vector: " + storageType);
+        }
+        throw new IllegalArgumentException("Unexpected storage type: " + storageType);
+    }
+
+    public static String toFileName(String pathOrInlineDv)
+    {
+        int randomPrefixLength = pathOrInlineDv.length() - Base85Codec.ENCODED_UUID_LENGTH;
+        String randomPrefix = pathOrInlineDv.substring(0, randomPrefixLength);
+        checkArgument(ALPHANUMERIC.matchesAllOf(randomPrefix), "Random prefix must be alphanumeric: %s", randomPrefix);
+        String prefix = randomPrefix.isEmpty() ? "" : randomPrefix + "/";
+        String encodedUuid = pathOrInlineDv.substring(randomPrefixLength);
+        UUID uuid = Base85Codec.decodeUuid(encodedUuid);
+        return "%sdeletion_vector_%s.bin".formatted(prefix, uuid);
+    }
+
+    public static ByteBuffer readDeletionVector(TrinoInputFile inputFile, int offset, int expectedSize)
+            throws IOException
+    {
+        byte[] bytes = new byte[expectedSize];
+        TrinoDataInputStream inputStream = new TrinoDataInputStream(inputFile.newStream());
+        inputStream.seek(offset);
+        int actualSize = inputStream.readInt();
+        if (actualSize != expectedSize) {
+            // TODO: Investigate why these size differ
+            log.warn("The size of deletion vector %s expects %s but got %s", inputFile.location(), expectedSize, actualSize);
+        }
+        inputStream.readFully(bytes);
+        return ByteBuffer.wrap(bytes).order(LITTLE_ENDIAN);
```

Comment:
in other places wre bytebuffer is constructed we don't have this, yet we rely on this.

i'd suggest ditching bytebuffers altogether. i find their API hard to use correctly. byte[] or Slice would be preferred.